### DISCUSSION TITLE
Module development tools

### DISCUSSION THREAD

**Discussion by svenoe** [RotherOSS/otobo#1614] [Projects]
It would be nice to at some point combine at least subsets of our different approaches of developing packages. This has various layers, e.g. do we want to be bound to an otobo installation or not, or docker vs native, etc. which are interesting aspects on this on their own - for starters though, let's begin by preparing a list of features which should/could be included.

- Linking of module files into the otobo environment
- As a more advanced addition to the above: "Install" packages from files in a directory rather than opm-packages; that is Linking the files as above, and performing all database changes in an sopm, as well as running potential var/packagesetup/ functions;
- Cleaning a system after development is done
- Preparing new files by adding copyright and an origin
- Provide a way to generate a sopm stub
- Provide a way to generate a documentation stub (I recently added a version of this to the core, but it probably would better be placed in a package: Kernel/System/Console/Command/Dev/Package/DocumentationGenerate.pm)
- Support the developer with setting code markers for changes in a file?
- Provide support for package updates if possible?

**Comment by bschmalhofer** [comment]
I second the notion to have the tools independent of an installation of OTOBO. It's also nice to have it as a CPAN module. Thus I propose to officially enhance and recommend https://metacpan.org/pod/OPM::Maker .
